lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -698,6 +753,52 @@ LogicalResult BuildOpGroups::buildOp(PatternRewriter &rewriter,
           yieldOp->getOpOperands());
   getState<ComponentLoweringState>().setWhileLoopLatchGroup(whileOpInterface,
                                                             assignGroup);
+  }
+  else if (auto ifOp = dyn_cast<scf::IfOp>(yieldOp->getParentOp())) {
+
+    ScfIfOp scfIfOp(ifOp);
+
+    bool isThenRegion = (yieldOp->getParentRegion() == &ifOp.getThenRegion());
+    bool isElseRegion = (yieldOp->getParentRegion() == &ifOp.getElseRegion());
+
+    Block *ifBlock = ifOp->getBlock();
+
+    if (isThenRegion) {
+      // create a calyx group for the then branch
+      std::string groupName = "then_group";
```

Comment:
You'll want to make sure there are no name clashes. See, e.g.,

https://github.com/llvm/circt/blob/307fb5874bfa7acd047f42a075c46c2ed3835599/lib/Conversion/CalyxToFSM/CalyxToFSM.cpp#L305-L310